src/module/flows/structure.ts

```diff
@@ -1,81 +1,472 @@
+      state.data.primary_roll,
+      "",
+      state.data.struct_lost,
+      rerollMacroData
+    );
+
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Handle logic for when a mech needs to perform a hull check after the initial structure roll
+ * @param state
+ * @returns
+ */
+export async function structureHullCheck(state: FlowState<LancerFlowState.StructureRollData>): Promise<boolean> {
+  if (!state.data) throw new TypeError(`Structure roll data flow state missing!`);
+
+  var actor = state.actor;
+  if (!actor.is_mech() && !actor.is_npc()) {
+    ui.notifications!.warn('Only npcs and mechs can work with "remaining structure" logic.');
+    return false;
+  }
+
+  if (state.data.primary_roll_result === 1 && state.data.struct_lost === 2) {
+    console.log("structure hull check criteria met");
+    let macroData = encodeMacroData({
+      title: "Hull",
+      fn: "prepareStatMacro",
+      args: [actor.uuid, "system.hull"],
+    });
+
+    let secondaryRoll = `<button class="chat-button chat-macro-button" data-macro="${macroData}"><i class="fas fa-dice-d20"></i> Hull</button>`;
+    readyAndPrintCommonStructureCard(state, secondaryRoll);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Handle logic for when a structure roll leads to the secondary structure roll
+ * @param state
+ * @returns
+ */
+export async function secondaryStructureRollCheck(
+  state: FlowState<LancerFlowState.StructureRollData>
+): Promise<boolean> {
+  if (!state.data) throw new TypeError(`Structure roll data flow state missing!`);
+
+  var actor = state.actor;
+  if (!actor.is_mech() && !actor.is_npc()) {
+    ui.notifications!.warn('Only npcs and mechs can work with "remaining structure" logic.');
+    return false;
+  }
+
+  if (state.data.primary_roll_result >= 2 && state.data.primary_roll_result <= 4) {
+    console.log("structure secondary roll criteria met");
+    state.data.secondary_roll_check = true;
+    let macroData = encodeMacroData({
+      // TODO: Should create a "prepareRollMacro" or something to handle generic roll-based macros
+      // Since we can't change prepareTextMacro too much or break everyone's macros
+      title: "Roll for Destruction",
+      fn: "beginSecondaryStructureFlow",
+      args: [state.actor.uuid!, state],
+    });
+
+    let secondaryRoll = `<button class="chat-macro-button"><a class="chat-button" data-macro="${macroData}"><i class="fas fa-dice-d20"></i> Destroy</a></button>`;
+    readyAndPrintCommonStructureCard(state, secondaryRoll);
+    return false;
+  }
+
+  return true;
+}
+
+function readyAndPrintCommonStructureCard(
+  state: FlowState<LancerFlowState.StructureRollData>,
+  secondaryRoll: string
+): boolean {
+  if (!state.data) throw new TypeError(`Structure roll data flow state missing!`);
+  let rerollMacroData = encodeMacroData({
+    title: "Structure Damage",
+    fn: "beginStructureFlow",
+    args: [state.actor.uuid!, { reroll_data: { structure: state.data.struct_lost } }],
```

Comment:
Alright, taking a look now. A question and a statement:

- Where can I find an example of the "Callback attached to the button" referenced above?
- I had the helper functions because half the flow step functions were printing out the structure card using the same template data object built from the values in the state. However, that wouldn't be needed if the state interface matched the fields in the template.